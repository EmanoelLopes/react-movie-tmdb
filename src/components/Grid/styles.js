import styled from 'styled-components';
import { media } from 'utils/styled';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: auto;
  grid-template-rows: auto; 
  grid-column-gap: 0;
  grid-row-gap: 25px;

  ${media.phone`
    grid-template-columns: 50% 50%;
    grid-column-gap: 5px;
  `}

  ${media.desktop`
    grid-template-columns: 24% 24% 24% 24%;
    grid-column-gap: 15px;
  `}

  ${media.largeDesktop`
    grid-template-columns: 19% 19% 19% 19% 19%;
  `}
`;
