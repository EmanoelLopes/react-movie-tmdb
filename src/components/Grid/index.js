import React from 'react';
import { node, string } from 'prop-types';
import { GridContainer } from './styles';

const Grid = props => {
  const {
    children,
    columns,
    rows,
    columnGap,
    rowGap,
  } = props;

  return (
    <GridContainer
      columns={columns}
      rows={rows}
      columnGap={columnGap}
      rowGap={rowGap}>
      {children}
    </GridContainer>
  );
};

Grid.propTypes = {
  children: node,
  columns: string,
  rows: string,
  columnGap: string,
  rowGap: string,
};

Grid.defaultProps = {
  children: {},
  columns: 'auto',
  rows: 'auto',
  columnGap: '25px',
  rowGap: '25px',
};

export default Grid;
