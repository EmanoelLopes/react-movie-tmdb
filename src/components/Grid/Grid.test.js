import React from 'react';
import { mount } from 'enzyme';
import Grid from 'components/Grid';

const Children = '<div></div>';

describe('<Grid />', () => {
  it('should match snapshot', () => {
    const wrapped = mount(<Grid>{Children}</Grid>);
    expect(wrapped).toMatchSnapshot();
  });
});
