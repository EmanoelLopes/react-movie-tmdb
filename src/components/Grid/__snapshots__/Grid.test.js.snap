// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> should match snapshot 1`] = `
.c0 {
  display: grid;
  grid-template-columns: auto;
  grid-template-rows: auto;
  grid-column-gap: 0;
  grid-row-gap: 25px;
}

@media (min-width:36em) {
  .c0 {
    grid-template-columns: 50% 50%;
    grid-column-gap: 5px;
  }
}

@media (min-width:62em) {
  .c0 {
    grid-template-columns: 24% 24% 24% 24%;
    grid-column-gap: 15px;
  }
}

@media (min-width:80em) {
  .c0 {
    grid-template-columns: 19% 19% 19% 19% 19%;
  }
}

<Grid>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: grid;
  grid-template-columns: auto;
  grid-template-rows: auto; 
  grid-column-gap: 0;
  grid-row-gap: 25px;

  ",
              "
    @media (min-width: ",
              "36",
              "em) {
      ",
              "
    grid-template-columns: 50% 50%;
    grid-column-gap: 5px;
  ",
              "
    }
  ",
              "

  ",
              "
    @media (min-width: ",
              "62",
              "em) {
      ",
              "
    grid-template-columns: 24% 24% 24% 24%;
    grid-column-gap: 15px;
  ",
              "
    }
  ",
              "

  ",
              "
    @media (min-width: ",
              "80",
              "em) {
      ",
              "
    grid-template-columns: 19% 19% 19% 19% 19%;
  ",
              "
    }
  ",
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        &lt;div&gt;&lt;/div&gt;
      </div>
    </StyledComponent>
  </styled.div>
</Grid>
`;
